services/src/projects/project.rs

```diff
@@ -53,16 +55,25 @@ impl Project {
             project.description = description;
         }
 
-        if let Some(layers) = update.layers {
-            for (i, layer) in layers.into_iter().enumerate() {
-                if let Some(layer) = layer {
-                    if i >= project.layers.len() {
-                        project.layers.push(layer);
-                    } else {
-                        project.layers[i] = layer;
-                    }
+        if let Some(layer_updates) = update.layers {
+            let layers = mem::replace(&mut project.layers, Vec::new()).into_iter();
+            let mut layer_updates = layer_updates.into_iter();
+
+            for (layer, layer_update) in layers.zip(&mut layer_updates) {
+                match layer_update {
+                    LayerUpdate::Keep => project.layers.push(layer),
+                    LayerUpdate::Update(updated_layer) => project.layers.push(updated_layer),
+                    LayerUpdate::Delete => {}
                 }
             }
+
```

Comment:
what about layers that are neither addressed as keep, update or delete? should get an error then too.

Reply:
This is an implicit deletion.

Reply:
could also be implicit Keep... Neither is really intuitive which is why I would propose to throw error. Otherwise we have to document this behavior.

Reply:
The advantage here is that I can just set an arbitrary layer list of arbitrary length (all updates) and it works in all cases.
I would rather document it this way.

Reply:
ok

Reply:
https://github.com/geo-engine/geoengine/commit/ec3fb0f6bac6bbf3e76339fc1ee4d2bf0c6f7c13#diff-81d6c97ee571fac79cd18261a47f66164451154aa8f637d1b60cbc9dacea210aR45